lib/topic_query.rb

```diff
@@ -637,9 +637,13 @@ def default_results(options = {})
     options.reverse_merge!(@options)
     options.reverse_merge!(per_page: per_page_setting) unless options[:limit] == false
 
-    # Whether to return visible topics
-    options[:visible] = true if @user.nil? || @user.regular?
-    options[:visible] = false if @user && @user.id == options[:filtered_to_user]
+    # Whether to include unlisted (visible = false) topics
+    viewing_own_topics = @user && @user.id == options[:filtered_to_user]
+
+    if options[:visible].nil?
+      options[:visible] = true if @user.nil? || @user.regular?
+      options[:visible] = false if @guardian.can_see_unlisted_topics? || viewing_own_topics
```

Comment:
Previously staff users would go through neither of these two conditional paths, and options[:visible] would remain nil, which still worked as intended because we rely on a falsiness check later.

With the introduction of #can_see_unlisted_topics?, options[:visible] will be set to false for staff users, which has the same effect but is more explicit.